### PR TITLE
Add scc

### PR DESCRIPTION
**Pull request by Suxiaocai**

### CODE REVIEW

File: algorithms/generated/scc.gsql [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+                               ACCUM t.@is_backward = TRUE;
+                END;
+          
+                active = SELECT s
+                         FROM active:s 
+                         ACCUM IF s.@is_forward == TRUE AND s.@is_backward == TRUE THEN  // scc
+                                     s.@detached = TRUE,
+                                     @@s_cid_map += (s.@cid -> s.@vid)
+                               END,
+                               IF s.@is_forward == TRUE THEN
```
Comment (xchang2014) [inline algorithms/generated/scc.gsql]: maybe we could simplify the if else structure?

if f == true and b == ture then
else if f == true
else if b == ture
end

File: algorithms/generated/scc.gsql [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+                               FROM active:s -((reverse_link_to|reverse_link_to2|reverse_link_to3):e) -> :t
+                               WHERE t.@detached == FALSE AND s.@cid == t.@cid
+                               ACCUM s.@has_pos_indegree = TRUE;
+                        trim_set = SELECT s
+                               FROM active:s
+                               WHERE s.@has_pos_indegree == FALSE OR s.@has_pos_outdegree == FALSE
+                               ACCUM s.@detached = TRUE,
+                                     s.@cid = s.@vid;
+                
+                        print trim_set.size();
```
Comment (xchang2014) [inline algorithms/generated/scc.gsql]: is this for debug?

File: algorithms/generated/scc.gsql [hunk 3]
```diff
@@ -0,0 +1,174 @@
+CREATE DISTRIBUTED QUERY scc (INT iter = 500, INT iter_wcc = 5, INT top_k_dist) FOR GRAPH baidu{ //INT iter_end_trim = 3
+/* This query detects strongly connected components based on the following papers:
+   https://www.sandia.gov/~apinar/papers/irreg00.pdf
+   https://www.sciencedirect.com/science/article/pii/S0743731505000535
+   https://stanford-ppl.github.io/website/papers/sc13-hong.pdf
+
+   iter: number of iteration of the algorithm
+   iter_end_trim: stop trimming from this iteration, usually a small number(3 to 10)
```
[the remark this answers is not in the document]
Comment (Suxiaocai) [reply algorithms/generated/scc.gsql]: yes. this is comment

File: algorithms/generated/scc.gsql (hunk 2 above)
Comment (Suxiaocai) [reply algorithms/generated/scc.gsql]: I want to show the user this information, then the user knows how the algorithm goes. Just like the modularity value for each iteration in Louvain

File: algorithms/generated/scc.gsql [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+   iter_wcc: find wcc in this iteration; usually a small number(3 to 10)
```
[the remark this answers is not in the document]
Comment (Suxiaocai) [reply algorithms/generated/scc.gsql]: I added some more explanation

**Comment by Suxiaocai** [comment]
@xchang2014 
1. Revised according to your comments
2. Fixed install.sh bug found during testing
3. generated script examples

**Pull request merged**